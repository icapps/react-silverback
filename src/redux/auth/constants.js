export default {
  LOGIN_USER: 'LOGIN_USER',
  LOGIN_USER_PENDING: 'LOGIN_USER_PENDING',
  LOGIN_USER_REJECTED: 'LOGIN_USER_REJECTED',
  LOGIN_USER_FULFILLED: 'LOGIN_USER_FULFILLED',
  LOGOUT_USER: 'LOGOUT_USER',
};
