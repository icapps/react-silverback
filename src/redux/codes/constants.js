export default {
  GET_LANGUAGE_CODES: 'GET_LANGUAGE_CODES',
  GET_LANGUAGE_CODES_PENDING: 'GET_LANGUAGE_CODES_PENDING',
  GET_LANGUAGE_CODES_FULFILLED: 'GET_LANGUAGE_CODES_FULFILLED',
  GET_LANGUAGE_CODES_REJECTED: 'GET_LANGUAGE_CODES_REJECTED',

  GET_LANGUAGE_CODE_BY_ID: 'GET_LANGUAGE_CODE_BY_ID',
  GET_LANGUAGE_CODE_BY_ID_PENDING: 'GET_LANGUAGE_CODE_BY_ID_PENDING',
  GET_LANGUAGE_CODE_BY_ID_FULFILLED: 'GET_LANGUAGE_CODE_BY_ID_FULFILLED',
  GET_LANGUAGE_CODE_BY_ID_REJECTED: 'GET_LANGUAGE_CODE_BY_ID_REJECTED',

  CREATE_LANGUAGE_CODE: 'CREATE_LANGUAGE_CODE',
  CREATE_LANGUAGE_CODE_PENDING: 'CREATE_LANGUAGE_CODE_PENDING',
  CREATE_LANGUAGE_CODE_FULFILLED: 'CREATE_LANGUAGE_CODE_FULFILLED',
  CREATE_LANGUAGE_CODE_REJECTED: 'CREATE_LANGUAGE_CODE_REJECTED',

  DEPRECATE_LANGUAGE_CODE: 'DEPRECATE_LANGUAGE_CODE',
  DEPRECATE_LANGUAGE_CODE_PENDING: 'DEPRECATE_LANGUAGE_CODE_PENDING',
  DEPRECATE_LANGUAGE_CODE_FULFILLED: 'DEPRECATE_LANGUAGE_CODE_FULFILLED',
  DEPRECATE_LANGUAGE_CODE_REJECTED: 'DEPRECATE_LANGUAGE_CODE_REJECTED',

  UNDEPRECATE_LANGUAGE_CODE: 'UNDEPRECATE_LANGUAGE_CODE',
  UNDEPRECATE_LANGUAGE_CODE_PENDING: 'UNDEPRECATE_LANGUAGE_CODE_PENDING',
  UNDEPRECATE_LANGUAGE_CODE_FULFILLED: 'UNDEPRECATE_LANGUAGE_CODE_FULFILLED',
  UNDEPRECATE_LANGUAGE_CODE_REJECTED: 'UNDEPRECATE_LANGUAGE_CODE_REJECTED',

  GET_STATUS_CODES: 'GET_STATUS_CODES',
  GET_STATUS_CODES_PENDING: 'GET_STATUS_CODES_PENDING',
  GET_STATUS_CODES_REJECTED: 'GET_STATUS_CODES_REJECTED',
  GET_STATUS_CODES_FULFILLED: 'GET_STATUS_CODES_FULFILLED',

  SET_CODES_SORT: 'SET_CODES_SORT',
};
