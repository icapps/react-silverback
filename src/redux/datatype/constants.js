export default {
  GET_DATA_TYPE: 'GET_DATA_TYPE',
  GET_DATA_TYPE_PENDING: 'GET_DATA_TYPE_PENDING',
  GET_DATA_TYPE_REJECTED: 'GET_DATA_TYPE_REJECTED',
  GET_DATA_TYPE_FULFILLED: 'GET_DATA_TYPE_FULFILLED',
};
