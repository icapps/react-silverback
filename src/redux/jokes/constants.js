export default {
  GET_JOKE           : 'GET_JOKE',
  GET_JOKE_REQUEST   : 'GET_JOKE_REQUEST',
  GET_JOKE_FULFILLED : 'GET_JOKE_FULFILLED',
  GET_JOKE_REJECTED  : 'GET_JOKE_REJECTED',
};
