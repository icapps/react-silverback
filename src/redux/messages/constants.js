export default {
  SET_MESSAGE: 'SET_MESSAGE',
  
  CLEAR_MESSAGES: 'CLEAR_MESSAGES',
};
