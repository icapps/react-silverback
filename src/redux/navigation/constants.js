export default {
  TOGGLE_NAVIGATION: 'TOGGLE_NAVIGATION',
};
