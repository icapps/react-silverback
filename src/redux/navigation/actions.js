import constants from './constants';

export const toggleNavigation = () => {
  return {
    type: constants.TOGGLE_NAVIGATION,
  };
};
