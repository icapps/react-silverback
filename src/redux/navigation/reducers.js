import constants from './constants';

const initialState = {
  isNavigationShown: false,
};

const navigation = (state = initialState, action = {}) => {
  switch (action.type) {
    case constants.TOGGLE_NAVIGATION:
      return {
        ...state,
        isNavigationShown: !state.isNavigationShown,
      };
    default:
      return state;
  }
};

export default navigation;
