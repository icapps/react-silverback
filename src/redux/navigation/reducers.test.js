import navigation from './reducers';
import constants from './constants';

const initialState = {
  isNavigationShown: false,
};

describe('navigation reducer', () => {
  it('should TOGGLE_NAVIGATION', () => {
    expect(
      navigation(initialState, {
        type: constants.TOGGLE_NAVIGATION,
      }),
    ).toMatchInlineSnapshot(`
      Object {
        "isNavigationShown": true,
      }
    `);
  });
});
