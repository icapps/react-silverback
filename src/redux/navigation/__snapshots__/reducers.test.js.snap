// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navigation reducer should TOGGLE_NAVIGATION 1`] = `
Object {
  "isNavigationShown": true,
}
`;

exports[`navigation reducer should return the initial state 1`] = `
Object {
  "isNavigationShown": false,
}
`;
